esp/templates/program/modules/programprintables/studentschedule.tex

```diff
@@ -1,88 +1,99 @@
- \Large \textbf{ User name: {{ student.username|texescape }} $\bullet$ ID: {{ student.id }} } \\
-\end{minipage}
-{% endspaceless %}
-&
-%\includegraphics[width=0.6in]{splogol}
-\vspace{0.25in}
+Student Schedule for \\
+
+\begin{spacing}{1}
+\LARGE \bfseries {{ student.name|texescape }}
+\end{spacing}
 
-\end{tabularx}
 \end{center}
 
+\columnbreak
+
+\begin{flushright}
+\barcodeblock
+\end{flushright}
 
+\end{multicols}
 
-\normalsize
+\renewcommand\tabularxcolumn[1]{m{#1}}
 
 \begin{center}
-\begin{tabularx}{17cm}{|X|p{1.5in}|m{3.63in}|}
+{\large \textbf{Class Schedule} \\}
+\vspace{0.25cm}
+\begin{tabularx}{\textwidth}{|m{1.7in}|X|m{0.55in}|}
 \hline
-\multicolumn{3}{c}{
-\rowstyle{\color{white}\cellcolor{black} } 
-\textbf{Class Schedule for {{ student.name|texescape }} } }  \\
-\multicolumn{1}{|c|}{\textbf{Class Time} } & \textbf{Room} & \multicolumn{1}{c|}{\textbf{Class} } \\
+\textbf{Time and Room} & \textbf{Class and Teacher} & \textbf{Code} \\
 {% for cls in student.classes %}\hline
-{% ifequal cls.event_type.description "Compulsory" %}
-  \multicolumn{1}{|c|}{ {{ cls.start|date:"f" }} -- {{ cls.end|date:"f A" }} } & \multicolumn{2}{|c|}{ \textbf{\textit{ {{ cls.description }} } } }
-{% else %}
-  {% ifequal cls.event_type.description "Class" %}
-  \multicolumn{1}{|c|}{ {{ cls.start|date:"f" }} -- {{ cls.end|date:"f A" }} } & \multicolumn{2}{c|}{[NO CLASS]}
-  {% else %}
-  \multicolumn{1}{|c|}{ {{ cls.time_blocks.0.start|date:"f" }} -- {{ cls.time_blocks.0.end|date:"f A" }} } & {% if cls.initial_rooms|length_is:0 %}N/A{% else %}{{ cls.prettyrooms|join:", "|texescape }}{% endif %} & {{ cls.emailcode }}: {{ cls.title|truncatewords:5|texescape }}
-  {% endifequal %}
-{% endifequal %}
-\\ 
+{% if cls.friendly_times|length_is:"0" %}N/A{% endif %}{{ cls.friendly_times|join:", " }} \newline
+\footnotesize{ {% if cls.initial_rooms|length_is:0 %}N/A{% else %}{{ cls.prettyrooms.0|texescape }}{% endif %} } &
+{% if cls.title|length_is:"0" %}N/A{% endif %} {{ cls.title|truncatewords_char:45|texescape }} \newline
+\footnotesize{ {{ cls.parent_class.getTeacherNames|join:", "|truncatewords_char:50|texescape }} }  &
+\texttt{% templatetag openbrace %}{{ cls.emailcode }}{% templatetag closebrace %} \\
 {% endfor %}\hline
 \end{tabularx}
-\end{center}
 
+\end{center}
 
-\begin{center}
-\begin{tabularx}{17cm}{X c}
-\multicolumn{2}{c}{\small
-\textit{Please see your map for building directions, or ask anyone for help.}
-\normalsize } \\
-\multicolumn{2}{c}{\small
-\textit{If you are not signed up for a full day of classes, we encourage you to add more!}
-\normalsize } \\
-~ & ~ 
-\end{tabularx}
+\begin{center}	
+\begin{tabularx}{17cm}{X c}	
+\multicolumn{2}{c}{\small	
+\textit{Please see your map for building directions, or ask anyone for help.}	
+\normalsize } \\	
+\multicolumn{2}{c}{\small	
+\textit{If you are not signed up for a full day of classes, we encourage you to add more!}	
+\normalsize } \\	
+~ & ~ 	
+\end{tabularx}	
```

Comment:
Can you get rid of all the extra spaces (or maybe they're tabs) at the end of these lines?

Reply:
This refers to all of lines 82-91... apparently github only shows 4 lines even though I highlighted all the lines.

Reply:
Fixed!